src/basic_recipes/poly.jl

```diff
@@ -59,14 +60,14 @@ function poly_convert(polygons::AbstractVector{<: AbstractVector{<: VecTypes}})
     end
 end
 
-to_line_segments(polygon) = convert_arguments(LineSegments, polygon)[1]
+to_line(polygon) = convert_arguments(Lines, polygon)[1]
```

Comment:
I renamed and switched to `Lines` because this gets pushed into `lines` further down the pipeline. Without the change here we get duplicate points which results in some ugly corners on this branch. (Because joins still get drawn for zero length lines I guess)

Something I else I noticed here is that `Rect` doesn't make it to `plot!(plot::Poly{<: Tuple{<: Union{Polygon, AbstractVector{<: PolyElements}}}})`. It uses `plot!(plot::Poly{<: Tuple{Union{GeometryBasics.Mesh, GeometryPrimitive}}})` instead. That results in in every `Rect` having 4 bad corners instead of just one... (That should also be the case on master) I tried adding `Rect` explicitly to the nicer `plot!` method but it still used the other. I don't know why